python/lsst/cp/pipe/linearity.py

```diff
@@ -192,75 +232,237 @@ def run(self, inputPtc, camera=None, inputDims=None):
+                numPerBin, binEdges = np.histogram(linearOrdinate, bins=self.config.fitOrder)
+                with np.errstate(invalid="ignore"):
+                    values = np.histogram(linearOrdinate, bins=self.config.fitOrder,
+                                          weights=(inputAbscissa[fluxMask] - linearOrdinate))[0]/numPerBin
+                    binCenters = np.histogram(linearOrdinate, bins=self.config.fitOrder,
+                                              weights=linearOrdinate)[0]/numPerBin
+                    values = values[numPerBin > 0]
+                    binCenters = binCenters[numPerBin > 0]
+
+                self.debugFit('splineFit', binCenters, np.abs(values), values, None, ampName)
+                interp = afwMath.makeInterpolate(binCenters.tolist(), values.tolist(),
+                                                 afwMath.stringToInterpStyle("AKIMA_SPLINE"))
+                modelAbscissa = linearOrdinate + interp.interpolate(linearOrdinate)
+                self.debugFit('splineFit', linearOrdinate, fitAbscissa, modelAbscissa, None, ampName)
+
+                # If we exclude a lot of points, we may end up with
+                # less than fitOrder points.  Pad out the low-flux end
+                # to ensure equal lengths.
+                if len(binCenters) != self.config.fitOrder:
+                    padN = self.config.fitOrder - len(binCenters)
+                    binCenters = np.pad(binCenters, (padN, 0), 'linear_ramp',
+                                        end_values=(binCenters.min() - 1.0, ))
+                    # This stores the correction, which is zero at low values.
+                    values = np.pad(values, (padN, 0))
+
+                # Pack the spline into a single array.
+                linearityFit = np.concatenate((binCenters.tolist(), values.tolist())).tolist()
+                polyFit = [0.0]
+                polyFitErr = [0.0]
+                chiSq = np.nan
             else:
                 polyFit = [0.0]
                 polyFitErr = [0.0]
                 chiSq = np.nan
                 linearityFit = [0.0]
 
-            linearizer.linearityType[ampName] = 'self.config.linearityType'
-            linearizer.linearityCoeffs[ampName] = linearityFit
+            linearizer.linearityType[ampName] = self.config.linearityType
+            linearizer.linearityCoeffs[ampName] = np.array(linearityFit)
             linearizer.linearityBBox[ampName] = amp.getBBox()
-            linearizer.fitParams[ampName] = polyFit
-            linearizer.fitParamsErr[ampName] = polyFitErr
+            linearizer.fitParams[ampName] = np.array(polyFit)
+            linearizer.fitParamsErr[ampName] = np.array(polyFitErr)
             linearizer.fitChiSq[ampName] = chiSq
-            self.debugFit('solution', timeVector, meanVector, linearizer, ampName)
 
+            image = afwImage.ImageF(len(inputOrdinate), 1)
+            image.getArray()[:, :] = inputAbscissa
```

Comment:
I get that they're the same length, I just found it odd to declare it with the length of `inputOrdinate` but then assign it to `inputAbscissa` instead (unless this is a very-late-in-the-day form of cross checking)